packages/hub/services/crank-nonce-lock.ts

```diff
@@ -0,0 +1,65 @@
+import { inject } from '@cardstack/di';
+import { Prisma } from '@prisma/client';
+import BN from 'bn.js';
+import config from 'config';
+import { addSeconds } from 'date-fns';
+import { Wallet } from 'ethers';
+import { nowUtc } from '../utils/dates';
+
+export default class CrankNonceLock {
+  prismaManager = inject('prisma-manager', { as: 'prismaManager' });
+  ethersProvider = inject('ethers-provider', { as: 'ethersProvider' });
+  CRANK_LOCK_KEY = 1;
+
+  private async getNonce(chainId: number, dbTransaction: Prisma.TransactionClient) {
+    let crankNonce = await dbTransaction.crankNonce.findUnique({
+      where: { chainId: chainId },
+    });
+    let one = new BN(1);
+    let nextNonce = crankNonce?.nonce ? new BN(crankNonce.nonce.toString()).add(one) : new BN(0);
+    let nonceTTL = Number(config.get('nonceTTL'));
+    let now = nowUtc();
+    if (nextNonce.cmp(one) <= 0 || (crankNonce && addSeconds(crankNonce.updatedAt, nonceTTL) <= now)) {
```

Comment:
I think `nextNonce.lte(one)` is a bit simpler to say

Reply:
updated